actionmailer/lib/action_mailer/base.rb

```diff
@@ -883,33 +887,33 @@ def default_i18n_subject(interpolations = {})
     end
 
     def collect_responses(headers) #:nodoc:
-      responses = []
-
       if block_given?
         collector = ActionMailer::Collector.new(lookup_context) { render(action_name) }
         yield(collector)
-        responses = collector.responses
+        collector.responses
       elsif headers[:body]
-        responses << {
+        [{
           body: headers.delete(:body),
           content_type: self.class.default[:content_type] || "text/plain"
-        }
+        }]
       else
-        templates_path = headers.delete(:template_path) || self.class.mailer_name
-        templates_name = headers.delete(:template_name) || action_name
+        collect_responses_from_templates(headers)
+      end
+    end
 
-        each_template(Array(templates_path), templates_name) do |template|
-          self.formats = template.formats
+    def collect_responses_from_templates(headers)
+      templates_path = headers.delete(:template_path) || self.class.mailer_name
```

Comment:
Turns out we can go without mutations here, `:template_path` and `:template_name` are explicitly ignored where the headers are assigned to message. Here, changed: https://github.com/hnatt/rails/commit/baaf3769ff179248353ced0c60495c349a912fe1